### DISCUSSION TITLE
Force static linking (System wide library)

### DISCUSSION THREAD

**Discussion by babaliaris** [premake/premake-core#1817] [Q&A] [answered]
Hello!

Is it possible to force static linking against a library?

For example:
```lua

links {
    "SDL2",
    "SDL2_image"
}
```

I have SDL installed system widely from my Linux Dist package manager, and the above always links against the .so (shared) library.
SDL also provides a static libary .a

Is there a way to tell premake5 to link against the static library instead?
Can't find something similar in the Docs... Or by googling it...

Thanks!

**Comment by nickclark2016** [comment, marked as answer]
I want to say that I remember being able to do something like this to force it to be static.
```lua
links {
    "SDL2:static",
    "SDL2_image:static"
}
```

That said, I don't know if that is still correct.  I cannot find this in unit testing, so I'd be cautious using it.  That said, this should probably be something that we support better.

**Reply by nickclark2016** [reply]
Just wanted to follow up and see if this resolved your problem.

**Reply by babaliaris** [reply]
Yes it worked! Why isn't this in the docs?

**Reply by KyrietS** [reply]
These decorators were introduced in 2017 when Premake stored documentation on GitHub Wiki. At the time, features were being added, but the documentation didn't keep up. Now we are already trying to document everything on a regular basis. Thankfully there are still people who possess and pass on our tribal knowledge, like @nickclark2016.

I've created an issue for adding `:static` and `:shared` to the docs: https://github.com/premake/premake-core/issues/1818

> I cannot find this in unit testing

Isn't it tested here: https://github.com/premake/premake-core/blob/master/tests/tools/test_gcc.lua#L783 ?

**Reply by nickclark2016** [reply]
> These decorators were introduced in 2017 when Premake stored documentation on GitHub Wiki. At the time, features were being added, but the documentation didn't keep up. Now we are already trying to document everything on a regular basis. Thankfully there are still people who possess and pass on our tribal knowledge, like @nickclark2016.
> 
> I've created an issue for adding `:static` and `:shared` to the docs: #1818
> 
> > I cannot find this in unit testing
> 
> Isn't it tested here: https://github.com/premake/premake-core/blob/master/tests/tools/test_gcc.lua#L783 ?

Good catch for where it was in testing. Figured it was there, I just couldn't find it for some reason.

**Reply by babaliaris** [reply]
That makes sense. If you'd like to know, I learned about premake and most of how to use it from [TheCherno](https://www.youtube.com/watch?v=sULV3aB2qeU). I think he's one of the top Youtubers who uses premake. I would still be using cmake if not for Cherno.
